plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -0,0 +1,230 @@
+            access_token=dict(type='str', required=True, no_log=True),
+            gitlab_group=dict(type='str', required=True),
+            gitlab_user=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            access_level=dict(type='str', required=False, choices=['guest', 'reporter', 'developer', 'maintainer', 'owner'])
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_PY_GITLAB:
+        module.fail_json(msg=missing_required_lib('python-gitlab', url='https://python-gitlab.readthedocs.io/en/stable/'), exception=GITLAB_IMP_ERR)
+
+    server_url = module.params['server_url']
+    access_token = module.params['access_token']
+    gitlab_group = module.params['gitlab_group']
+    gitlab_user = module.params['gitlab_user']
+    state = module.params['state']
+    access_level = module.params['access_level']
+
+    # convert access level string input to int
+    if access_level:
+        access_level_int = {
+            'guest': gitlab.GUEST_ACCESS,
+            'reporter': gitlab.REPORTER_ACCESS,
+            'developer': gitlab.DEVELOPER_ACCESS,
+            'maintainer': gitlab.MAINTAINER_ACCESS,
+            'owner': gitlab.OWNER_ACCESS
+        }
+
+        access_level = access_level_int[access_level]
+
+    # connect to gitlab server
+    try:
+        gl = gitlab.Gitlab(server_url, private_token=access_token)
+    except (gitlab.exceptions.GitlabAuthenticationError, gitlab.exceptions.GitlabGetError) as e:
+        module.fail_json(msg="Failed to authenticate to GitLab Server: %s: %s" % (server_url, e))
+
+    group = GitLabGroup(module, gl)
+
+    user_exists = group.does_user_exist(gitlab_user)
+    group_exists = group.does_group_exist(gitlab_group)
+
+    # group doesn't exist
+    if not group_exists:
+        module.fail_json(msg="group '%s' not found." % gitlab_group)
+
+    # user doesn't exist
+    if not user_exists:
+        if state == 'absent':
+            module.exit_json(changed=False, result="user '%s' not found, and thus also not part of the group" % gitlab_user)
+        else:
+            module.fail_json(msg="user '%s' not found." % gitlab_user)
+
+    # get group id
+    gitlab_group_id = group_exists[0].id
+
+    # get user id
+    gitlab_user_id = user_exists[0].id
+    is_user_a_member = group.is_user_a_member(gitlab_group_id, gitlab_user_id)
+
+    # check if the user is a member in the group
+    if not is_user_a_member:
+        if state == 'present':
+
+            if not access_level:
+                module.fail_json(msg="Access level must be entered for adding the member, '%s', to the GitLab group, '%s'." % (gitlab_user, gitlab_group))
+
+            # add user to the group
+            if not module.check_mode:
+                group.add_member_to_group(gitlab_user_id, gitlab_group_id, access_level)
+                module.exit_json(changed=True, result="Successfully added user '%s' to the group." % gitlab_user)
```

Comment:
exit_json **must** always be executed, not only in check mode. Please move it one level up (as in my suggestion).

(Same below.)